lib/bap_image/bap_image.ml

```diff
@@ -194,19 +200,23 @@ let validate_segments data (s,ss) : Validate.t =
 let is_virtual s = Backend.Segment.off s < 0
 
 let create_segments arch data (s,ss) =
+  let cmp_addr x y =
+    let addr s = Location.addr @@ Backend.Segment.location s in
+    Addr.compare (addr x) (addr y) in
   let endian = Arch.endian arch in
   (* a segment with the negative offset is virtual, we can't represent
      them in the Image, since the are not backed by static memory. But
      we will keep them, for analyses that require them.  *)
-  List.filter (s::ss) ~f:(Fn.non is_virtual) |> function
+  List.filter (s::ss) ~f:(Fn.non is_virtual) |>
```

Comment:
We will get rid of virtual segments, they are not needed anymore, but you can keep this code so far.